Visibility: support matching on target names. (#18369)
pantsbuild/pants: docs/markdown/Using Pants/validating-dependencies.md

Glob syntax
-----------

> The visibility rules is all about matching globs. There are two primary kinds of globs supported, the simple glob syntax and the path glob syntax. Simple globs only supports `*` as a match anything and is otherwise case sensitive and are used for all glob values except for the `path` part of a __target rule spec__. The path glob syntax is described further below.

Reviewer: This will seem pretty complicated to the user. Why not have one kind of glob, where `*` matches any character except `/` and `**` matches any character including `/` (which, IIUC, is what they mean in the current code).  Since non-paths won't have `/` in them, you can just use either, up to you.

Or do we really need two kinds of glob, for reasons I don't see?

> The visibility rules is all about matching globs. There are two primary kinds of globs supported, the simple glob syntax and the path glob syntax. Simple globs only supports `*` as a match anything and is otherwise case sensitive and are used for all glob values except for the `path` part of a __target rule spec__. The path glob syntax is described further below.

Author: No, you're right. As I'm trying to consolidate the two we're close enough now to start treating them as the same documentation wise (and soon also implementation wise).

> Path globs have two wildcards, the `*` as a match anything non-recursively and the `**` for a recursive match anything. All globs are matched until the end of the value it is being applied to, so if there is no trailing wildcard (`*` or `**`) the end of the path glob it will match until the end of the path. This allows for matching on file names regardless of where in the project tree they are:

Reviewer: ```suggestion
Path globs have two wildcards, the `*` to match anything non-recursively and the `**` to match anything recursively. Each glob is matched until the end of the value it is being applied to, so if there is no trailing wildcard (`*` or `**`) on the end of the path glob, it will match to the end of the path. This allows for matching on file names regardless of where in the project tree they are:
```

Glob syntax
-----------

The visibility rules are all about matching globs. There are two wildcards, the `*` to match anything except `/` and the `**` to match anything including any `/`. (For paths that is non-recursive and recursive globbing respectively.)

> Globs are matched until the end of the value it is being applied to, so if there is no trailing wildcard (`*` or `**`) on the end of the path glob, it will match to the end of the value. An example where this is usefull is for matching on file names regardless of where in the project tree they are:

Reviewer: ```suggestion
A glob is matched until the end of the value it is being applied to, so if there is no trailing wildcard (`*` or `**`) on the end of the path glob, it will match to the end of the value. An example where this is useful is for matching on file names regardless of where in the project tree they are:
```
